File: 8958.py
# 8958 py

n = int(input())
arr = []
point = 0
bonus = 0

for _ in range(n):
    arr.append(input())
for i in arr:
    for j in i:
        if j == 'O':
            point += 1 + bonus
            bonus += 1
        else:
            bonus = 0
    print(point)
    point = 0
    bonus = 0

File: 8958.c
// 8958 C

#include <stdio.h>

int    main(void)
{
	int		n;
	char	arr[80];
	int		bonus = 0;
	int		point = 0;

	scanf("%d", &n);
	for (int i = 0; i < n; i++)
	{
		scanf("%s", arr);
		for (int j = 0; arr[j]; j++)
		{
			if (arr[j] == 'O')
				point += 1 + bonus++;
			else
				bonus = 0;
		}
	printf("%d\n", point);
	point = 0;
	bonus = 0;
	}
}

